.cursor/rules/temporal-project-structure.mdc

```diff
@@ -0,0 +1,75 @@
+---
+description: "Defines and enforces standardized directory structure and documentation requirements for Temporal-based workflows and workers"
+globs: ["workers/**/*", "docs/user-guide/temporal/**/*"]
+alwaysApply: false
+---
+# Temporal Project Structure Rule
+
+## Purpose
+Defines the required directory and documentation structure for Temporal-based workflows and workers in this repository.
+
+## Project Structure
+
+The project follows a modular structure with workers as independent packages:
+
+```
+.
+├── workers/                  # Root directory for all Temporal workers
+│   ├── main/                 # Main worker package
+│   │   ├── src/              # Source code
+│   │   │   ├── activities/   # Activity implementations
+│   │   │   ├── workflows/    # Workflow definitions
+│   │   │   ├── index.ts      # Worker entry point
+│   │   │   └── types.ts      # Worker-specific types
+│   │   ├── package.json      # Worker dependencies
+│   │   ├── tsconfig.json     # TypeScript configuration
+│   │   └── README.md         # Worker documentation
+│   └── [other-workers]/      # Additional worker packages
+├── workers-shared/           # Shared utilities, types, and configuration for all workers
```

Comment:
honestly saying workers-shared/ sounds confusing. Very close to shared runners. Let's call it just common/ ?